pkg/es/config/config_test.go

```diff
@@ -0,0 +1,740 @@
+				mockMetricsFactory := metrics.NullFactory
+				config := &Configuration{
+					Servers:               []string{testServer.URL},
+					AllowTokenFromContext: true,
+					Password:              "secret",
+					LogLevel:              "info",
+					Username:              "user",
+					PasswordFilePath:      "",
+					BulkSize:              -1, // disable bulk; we want immediate flush
+				}
+				return config, mockLogger, mockMetricsFactory
+			},
+			expectedError: false,
+		},
+		{
+			name: "success with valid configuration and error log level",
+			setup: func() (*Configuration, *zap.Logger, metrics.Factory) {
+				testServer := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+					require.Equal(t, http.MethodGet, req.Method)
+					res.WriteHeader(http.StatusOK)
+					res.Write(mockEsServerResponseWithVersion0)
+				}))
+				mockLogger := zap.NewNop()
+				mockMetricsFactory := metrics.NullFactory
+				config := &Configuration{
+					Servers:               []string{testServer.URL},
+					AllowTokenFromContext: true,
+					Password:              "secret",
+					LogLevel:              "error",
+					Username:              "user",
+					PasswordFilePath:      "",
+					BulkSize:              -1, // disable bulk; we want immediate flush
+				}
+				return config, mockLogger, mockMetricsFactory
+			},
+			expectedError: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			config, logger, metricsFactory := test.setup()
+			client, err := NewClient(config, logger, metricsFactory)
+			fmt.Println(client)
+			fmt.Println(err)
+			if test.expectedError {
+				require.Error(t, err)
+				require.Nil(t, client)
+			} else {
+				require.NoError(t, err)
+				require.NotNil(t, client)
+			}
+		})
+	}
+}
+
+func TestApplyDefaults(t *testing.T) {
+	source := &Configuration{
+		RemoteReadClusters:           []string{"cluster1", "cluster2"},
+		Username:                     "sourceUser",
+		Password:                     "sourcePass",
+		Sniffer:                      true,
+		MaxSpanAge:                   100,
+		AdaptiveSamplingLookback:     50,
+		NumShards:                    5,
+		NumReplicas:                  1,
+		PrioritySpanTemplate:         10,
+		PriorityServiceTemplate:      20,
+		PriorityDependenciesTemplate: 30,
+		BulkSize:                     1000,
+		BulkWorkers:                  10,
+		BulkActions:                  100,
+		BulkFlushInterval:            30,
+		SnifferTLSEnabled:            true,
+		Tags:                         TagsAsFields{AllAsFields: true, DotReplacement: "dot", Include: "include", File: "file"},
+		MaxDocCount:                  10000,
+		LogLevel:                     "info",
+		SendGetBodyAs:                "json",
+	}
+
+	tests := []struct {
+		name     string
+		target   *Configuration
+		expected *Configuration
+	}{
+		{
+			name: "All Defaults Applied except PriorityDependenciesTemplate",
+			target: &Configuration{
```

Comment:
There is no need to assign default values for primitive fields, skip them and only assign what matters.

Reply:
resolved now